File: Sorts/InsertionSort/insertionSort.js
function insertionSort(arr) {
    const len = arr.length;
    for (let i = 0; i < len; i++) {
        const temp = arr[i];
        let j = i;
        while (j > 0 && arr[j - 1] >= temp) {
            arr[j] = arr[j - 1];
            j--;
        }
        arr[j] = temp;
    }
    return arr;
}

const array = [64, 34, 25, 12, 22, 11, 90];
insertionSort(array);
console.log("Sorted array: ", array);

File: Sorts/InsertionSort/InsertionSort.java
package Sorts.InsertionSort;

public class InsertionSort {
    public static void insertionSort(int[] arr) {
        int len = arr.length;
        for (int i = 1; i < len; i++) {
            int temp = arr[i];
            int j = i;
            while(j  > 0 && arr[j - 1] >= temp) {
                arr[j] = arr[j - 1];
                j--;
            }
            arr[j] = temp;
        }
    }

    public static void main(String[] args) {
        int[] arr = {64, 34, 25, 12, 22, 11, 90};
        insertionSort(arr);
        System.out.println("Sorted array:");
        for (int num : arr) {
            System.out.print(num + " ");
        }
    }
}
